vortex-serde/src/layouts/read/builder.rs

```diff
@@ -70,8 +72,14 @@ impl<R: VortexReadAt> LayoutReaderBuilder<R> {
         self
     }
 
+    pub fn with_message_cache(mut self, message_cache: Arc<RwLock<LayoutMessageCache>>) -> Self {
+        self.message_cache = Some(message_cache);
+        self
+    }
```

Comment:
I don't think we should externalize just yet. Imho we should only do it once there's a cache manager or some on disk impl